app/controllers/hourly_rates_controller.rb

```diff
@@ -83,8 +83,14 @@ def update
       return deny_access unless User.current.admin?
     end
 
-    @user.add_rates(@project, permitted_params.user_rates[:new_rate_attributes])
-    @user.set_existing_rates(@project, permitted_params.user_rates[:existing_rate_attributes])
+    if params.include? 'user'
+      update_rates @user,
+                   @project,
+                   permitted_params.user_rates[:new_rate_attributes],
+                   permitted_params.user_rates[:existing_rate_attributes]
+    else
+      delete_rates @user, @project
```

Comment:
The UI makes me expect that a specific rate is being deleted instead of all the rates.